code/game/objects/items/mail.dm

```diff
@@ -332,3 +334,166 @@
 /obj/item/paper/fluff/junkmail_generic/Initialize(mapload)
 	default_raw_text = pick(GLOB.junkmail_messages)
 	return ..()
+
+// =====================
+// Antag stuff down here
+// =====================
+
+/obj/item/mail/traitor
+	var/armed = FALSE
+	var/mob/madeby
+	goodie_count = 0
+
+/obj/item/mail/traitor/envelope
+	name = "envelope"
+	icon_state = "mail_large"
+	stamp_max = 2
+	stamp_offset_y = 5
+
+/obj/item/mail/traitor/attack_self(mob/user)
+	if(recipient_ref)
+		var/datum/mind/recipient = recipient_ref.resolve()
+		// If the recipient's mind has gone, then anyone can open their mail
+		// whether a mind can actually be qdel'd is an exercise for the reader
+		if(recipient && recipient != user?.mind)
+			to_chat(user, span_notice("You can't open somebody else's mail! That's <em>illegal</em>!"))
+			return
+
+	to_chat(user, span_notice("You start to unwrap the package..."))
+	if(!do_after(user, 1.5 SECONDS, target = user))
```

Comment:
All of this code should not be copied and pasted from normal mail. What is going to happen is someone is going to change normal mail, and then suddenly there is a traitor tell. Move these to a shared proc.

Reply:
Okay.